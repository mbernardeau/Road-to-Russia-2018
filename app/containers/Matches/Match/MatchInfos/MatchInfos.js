import React from 'react'
import PropTypes from 'prop-types'

import moment from 'moment'
import Tooltip from 'material-ui/Tooltip'

import StadiumTooltip from './StadiumTooltip'

const containerStyles = {
  marginTop: '10px',
  color: 'darkgray',
  display: 'flex',
  justifyContent: 'space-around',
  fontSize: '0.9em',
  textShadow: '0 1px 0 rgba(0,0,0,.1)',
}

const MatchInfos = ({ match }) => {
  const dateTime = moment.unix(match.dateTime)

  return (
    <div style={containerStyles}>
      <Tooltip title={dateTime.format('LLL')} disableTriggerTouch>
        <div>{dateTime.fromNow()}</div>
      </Tooltip>
      <div>•</div>
      <Tooltip title={<StadiumTooltip stadium={match.stadium} />} disableTriggerTouch>
        <div>{match.stadium.name}</div>
      </Tooltip>
      <div>•</div>
      <Tooltip
        title={<StadiumTooltip stadium={match.stadium} />}
        placement="left"
        disableTriggerTouch
      >
        <div>{match.stadium.city}</div>
      </Tooltip>
    </div>
  )
}

MatchInfos.propTypes = {
  match: PropTypes.shape({
    dateTime: PropTypes.number.isRequired,
    stadium: PropTypes.shape({
      name: PropTypes.string.isRequired,
      city: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired,
}

export default MatchInfos
