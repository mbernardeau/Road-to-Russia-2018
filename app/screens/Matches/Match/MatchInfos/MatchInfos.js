import React from 'react'
import PropTypes from 'prop-types'

import moment from 'moment'
import Tooltip from 'material-ui/Tooltip'

import StadiumTooltip from './StadiumTooltip'

const containerStyles = {
  marginTop: '10px',
  color: 'darkgray',
  display: 'flex',
  justifyContent: 'space-around',
  fontSize: '0.9em',
  textShadow: '0 1px 0 rgba(0,0,0,.1)',
}

const MatchInfos = ({ match, stadium }) => {
  const dateTime = moment(match.dateTime)

  return (
    <div style={containerStyles}>
      <Tooltip title={dateTime.format('LLL')} disableTriggerTouch>
        <div>{dateTime.fromNow()}</div>
      </Tooltip>
      <div>•</div>
      <Tooltip title={<StadiumTooltip {...stadium} />} disableTriggerTouch>
        <div>{stadium.name}</div>
      </Tooltip>
      <div>•</div>
      <Tooltip title={<StadiumTooltip {...stadium} />} placement="left" disableTriggerTouch>
        <div>{stadium.city}</div>
      </Tooltip>
    </div>
  )
}

MatchInfos.defaultProps = {
  match: {
    dateTime: 0,
  },
  stadium: {},
}

MatchInfos.propTypes = {
  match: PropTypes.shape({
    dateTime: PropTypes.number.isRequired,
  }),
  stadium: PropTypes.shape({
    name: PropTypes.string,
    city: PropTypes.string,
  }),
}

export default MatchInfos
