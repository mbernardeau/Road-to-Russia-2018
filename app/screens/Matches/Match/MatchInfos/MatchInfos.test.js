import React from 'react'
import renderer from 'react-test-renderer'
import moment from 'moment'

import MatchInfos from './MatchInfos'

describe('screens/Match/StadiumTooltip', () => {
  // Mock date to ensure consistancy between tests
  Date.now = jest.fn(() => 1482363367071)

  moment.locale('fr')

  // Mock toLocaleString because of inconsistant behaviour in node (it works correctly in browsers)
  // eslint-disable-next-line
  Number.prototype.toLocaleString = function() {
    return `${this}`
  }

  const props = {
    match: {
      dateTime: 1529002800000,
    },
    stadium: {
      capacity: 69500,
      city: 'Saint-PÃ©tersbourg',
      name: 'Stade Krestovski',
    },
    matchId: 'matchid1',
  }

  it('renders correctly', () => {
    const tree = renderer.create(<MatchInfos {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
