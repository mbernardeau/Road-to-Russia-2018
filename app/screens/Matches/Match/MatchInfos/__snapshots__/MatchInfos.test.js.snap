// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screens/Match/StadiumTooltip renders correctly 1`] = `
<div
  className="match-infos-container"
>
  <Tooltip
    disableTriggerTouch={true}
    title="14 juin 2018 21:00"
  >
    <div>
      dans un an
    </div>
  </Tooltip>
  <div>
    •
  </div>
  <Tooltip
    disableTriggerTouch={true}
    title={
      <StadiumTooltip
        capacity={69500}
        city="Saint-Pétersbourg"
        name="Stade Krestovski"
      />
    }
  >
    <div>
      Stade Krestovski
    </div>
  </Tooltip>
  <div>
    •
  </div>
  <Tooltip
    disableTriggerTouch={true}
    placement="left"
    title={
      <StadiumTooltip
        capacity={69500}
        city="Saint-Pétersbourg"
        name="Stade Krestovski"
      />
    }
  >
    <div>
      Saint-Pétersbourg
    </div>
  </Tooltip>
</div>
`;
